curvefs/src/client/s3/client_s3_cache_manager.cpp

```diff
@@ -1396,8 +1425,12 @@ CURVEFS_ERROR DataCache::Flush(uint64_t inodeId, bool force) {
                 ret = s3ClientAdaptor_->GetDiskCacheManager()->Write(
                     objectName, data + writeOffset, n);
             } else {
-                ret = s3ClientAdaptor_->GetS3Client()->Upload(
-                    objectName, data + writeOffset, n);
+                auto context = std::make_shared<PutObjectAsyncContext>();
```

Comment:
while loop in line 1389 is unnecessary, you can just generate all upload contexts in here and store into a vector, and upload them out current while loop